catalog/solr/catalog-solr-offline-gazetteer/src/main/java/ddf/catalog/solr/offlinegazetteer/SyncCatalogCommand.java

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.solr.offlinegazetteer;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.filter.impl.SortByImpl;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.util.impl.ResultIterable;
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import net.jodah.failsafe.Failsafe;
+import net.jodah.failsafe.RetryPolicy;
+import org.apache.karaf.shell.api.action.Action;
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.lifecycle.Reference;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.apache.karaf.shell.api.console.Session;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.codice.ddf.security.Security;
+import org.codice.solr.client.solrj.SolrClient;
+import org.codice.solr.client.solrj.UnavailableSolrException;
+import org.codice.solr.factory.SolrClientFactory;
+import org.opengis.filter.sort.SortOrder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Service
+@Command(
+  scope = "offline-solr-gazetteer",
+  name = "synccatalog",
+  description = "Syncs all catalog items to the offline solr gazetteer core"
+)
+public class SyncCatalogCommand implements Action {
+
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(RemoveAllOfflineSolrGazetteerCommand.class);
```

Comment:
✏️   wrong class - should be `SyncCatalogCommand.class`

Reply:
Good, I am not the only one who makes this mistake.  😉

Reply:
@pklinef when he caught this i immediately thought, "You know, maybe we *do* do this on purpose....." 😂

Reply:
Maybe it is time to figure out how to make an Intellij macro to do this for me instead of copy and paste.

Reply:
I actually had an intellij Live Template for this, but when i upgraded computers or something long ago I lost it. time to go do it again. I'll post here if/when i get it figured

Reply:
@pklinef 
![image](https://user-images.githubusercontent.com/9013416/86950123-94480800-c104-11ea-84a2-5e0b2b6fd773.png)